exonum/src/blockchain/mod.rs

```diff
@@ -220,24 +223,24 @@ impl BlockchainMut {
     }
 
     /// Creates and commits the genesis block with the given genesis configuration.
-    // TODO: extract genesis block into separate struct [ECR-3750]
-    fn create_genesis_block(
-        &mut self,
-        config: ConsensusConfig,
-        initial_services: Vec<InstanceConfig>,
-    ) -> Result<(), Error> {
-        config.validate()?;
+    fn create_genesis_block(&mut self, genesis_config: GenesisConfig) -> Result<(), Error> {
+        genesis_config.consensus_config.validate()?;
         let mut fork = self.fork();
         // Write genesis configuration to the blockchain.
-        Schema::new(&fork).consensus_config_entry().set(config);
+        Schema::new(&fork)
+            .consensus_config_entry()
+            .set(genesis_config.consensus_config);
+
+        for ArtifactSpec { artifact, payload } in genesis_config.artifacts {
+            Dispatcher::commit_artifact(&fork, artifact.clone(), payload.clone())?;
+            if !self.dispatcher.is_artifact_deployed(&artifact) {
```

Comment:
Possible that this check is redundant, can we deploy artifact before this method call?

Reply:
Most probably it is - the uniqueness for an artifact is checked during the addition of artifact to the genesis config.

Reply:
I think we'll fix it later.

Reply:
Removed